### ISSUE TITLE
Error detected while processing function apexExecuteSnippet#run..<SNR>60_executeSoqlQuery:

### ISSUE THREAD

**Issue by micheal-w-wells** [neowit/vim-force.com#41]
I'm getting this with commit efd6bfe915bd5e152ff10de1e2800 and tooling-force.com-0.3.3.0.jar on any query (even "select id from case limit 1").

**Comment by neowit** [comment]
What is your OS, Vim version, do you use  tooling-force.com.jar in server or legacy mode?

Can you give a step-by-step description of how you are trying to run the query?

You referenced a specific commit, does it mean that before that commit you did not have this issue and could call `:ApexQuery` successfully?

**Comment by neowit** [comment]
One more thing - before all of the above - can you try to delete your local version of fim-force.com plugin and clone current version again. Just to make sure that it is not something to do with your local version being corrupted or something.

**Comment by micheal-w-wells** [comment]
Alright, so I just deleted my local version and cloned the latest.  Still the same issue.  I did mention that commit however I can't verify whether or not it worked before (I just wanted to ensure you knew I was using the newest version).  This is the first time I have tried to use the SOQL functionality.  

I'm running:
- Debian GNU/Linux 8.0 (jessie).  
- Vim 7.4

I believe I am running it in legacy mode (I haven't done anything to enable server mode).  

Basically in one of my projects/sandboxes I crack open any class file, type out the following query, highlight it and :ApexQuery and receive the following output:

QUERY on line 14:  select id from case limit 1
ERROR:  


No errors found
Error detected while processing function apexExecuteSnippet#run..<SNR>60_executeSoqlQuery:
line   17:
E37: No write since last change (add ! to override)
Press ENTER or type command to continue

**Comment by neowit** [comment]
Thank you for the update.

I believe the issue is that your .vimrc config is missing `set autowrite`, hence you are getting
> E37: No write since last change (add ! to override)  

this, in turn, leads to the error you originally reported.

If I am right then your choices are
- Either: save file before calling `:ApexQuery`
- OR: `set autowrite` so vim could save file for you

**Comment by micheal-w-wells** [comment]
That was it, thanks!
